Source/CarthageKit/Version.swift

```diff
@@ -30,7 +31,17 @@ public struct SemanticVersion: VersionType, Comparable {
 	///
 	/// Increments to this component represent backwards-compatible bug fixes.
 	public let patch: Int
-
+	
+	/// The pre-release identifier
+	///
+	/// Indicates that the version is unstable
+	public let preRelease : String?
```

Comment:
I'm wondering if there should be a inner `PreRelease` struct to use for this.

```swift
public struct SemanticVersion: VersionType, Comparable {
    public struct PreRelease: Comparable {
        public let identifiers: [String]
    }
}
```

Then a bunch of the code below (like `isPreReleaseLesser`) would have a more natural home.